### PR TITLE
Track donor name information when backing fields aren't in template

### LINKED ISSUE #6306: Gift Entry Payments: Contact First & Last (or Account Name) required on template layout in order to process an Elevate payment
Description: Not able to process payments when Contact First & Last (or Account Name)  are not added to the template.

As a gift entry user, I want to be able to use a streamlined template that only contains Contact 1 Lookup to process payments, so that I don't have to tab through extra sections/fields so that I can enter payments with maximum efficiency

Contact 1

Given an org that is an Elevate customer and that uses Gift Entry

And given a gift entry template that contains the Contact 1 lookup but does not contain Contact First & Last (and contains the Elevate widget)

When a user completes the Contact 1 lookup and enters payment information and submits the form

Then the contact information is able to be successfully sent to Elevate and the payment is able to be successfully processed

Account 1

Given an org that is an Elevate customer and that uses Gift Entry

And given a gift entry template that contains the Account 1 lookup but does not contain Account Name (and contains the Elevate widget)

When a user completes the Account 1 lookup and enters payment information and submits the form

Then the contact information is able to be successfully sent to Elevate and the payment is able to be successfully processed

### PR DESCRIPTION
**Pull request by bdvorachek**
WI: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008x8O3IAI/view

# Critical Changes

# Changes

# Issues Closed
Closes #6306 
# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 1]
```diff
@@ -2400,4 +2406,23 @@ export default class GeFormRenderer extends LightningElement{
...
+        const cardholderNameFields = [CONTACT_LAST_NAME_INFO, CONTACT_FIRST_NAME_INFO, ACCOUNT_NAME_FIELD];
+        return cardholderNameFields.find( f => apiNameFor(f) === fieldMapping.Target_Field_API_Name);
+    }
+
+    updateCardholderState(valueObjectFromRecord, fieldMapping) {
+        const value = this.getFieldValueForFormState(valueObjectFromRecord, fieldMapping);
+        this.cardholderNamesNotInTemplate[fieldMapping.Source_Field_API_Name] = value;
+    }
+
+    getFieldValueFromCardholderState(fieldApiNameOrFieldReference) {
```
Comment (voduyemi) [inline src/lwc/geFormRenderer/geFormRenderer.js:2420]: The first four lines of the function are duplicated with the first four lines in `getFieldValueFromFormState`. I think that's an opportunity for refactor or re-use. I'm also not sure we need to check for `fieldApiName` or `fieldReference` here, this function is called once and at that point, we know we know what we're passing to it (a field reference).

**Review by voduyemi — CHANGES_REQUESTED** [review]
Thanks! @bdvorachek. I've got a few comments around renaming variables and a potential refactor. Let me know what you think!

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 2]
```diff
@@ -973,22 +974,26 @@ export default class GeFormRenderer extends LightningElement{
...
-            ),
-            lastName: this.getFieldValueFromFormState(
-                DATA_IMPORT_CONTACT1_LASTNAME_FIELD
-            ),
-            accountName: this.getFieldValueFromFormState(
-                DATA_IMPORT_ACCOUNT1_NAME
-            )
+        let donorNames = {};
+
+        const nameFields = {
```
Comment (kenlewis-sfdo) [inline src/lwc/geFormRenderer/geFormRenderer.js:984]: What do you think about this donorNames function returning an instance of `NameFields` that exposes methods for firstName(), lastName(), and accountName()? (each returning a value that is from either formState or cardholder state)

Comment (psadie-sfdo) [reply src/lwc/geFormRenderer/geFormRenderer.js:984]: Really like this idea. We would pass in both state objects in the `NameFields` constructor?

**Comment by psadie-sfdo** [comment]
Thanks for this work Ben! Nothing to add at this point in addition to the above comments except to second the `NameFields` object.

**Review by kenlewis-sfdo — APPROVED** [review]
Looks great, Ben! Great to see the demo today also.  Thanks for this work!

**Review by dfuller114 — APPROVED** [review]
took a look at this one also. looks good to me! Great work Ben!

**Review by voduyemi — APPROVED** [review]
My comments are not blocking. Looks good to me @bdvorachek. Thanks!

**Review by psadie-sfdo — APPROVED** [review]
Great work! Looks good and looking forward to those additional jest tests!

**Pull request merged**